components/template-meta.tpl

```diff
@@ -2,7 +2,7 @@
 {% comment %}https://developers.facebook.com/tools/debug - Debug after each modification{% endcomment %}
 {% if site.data.fb_admin %}<meta property="fb:admins" content="{{ site.data.fb_admin }}">{% endif %}
 <meta property="og:type" content="{% if article %}article{% else %}website{% endif %}">
-<meta property="og:url" content="{{ site.url }}{% if article %}{{ article.url | remove_first:'/' }}{% else %}{{ page.url | remove_first:'/' }}{% endif %}">
+<meta property="og:url" content="{{ site.url }}{% if article %}{{ article.url | remove_first:'/' }}{% elsif product %}{{ product.url | remove_first:'/' }}{% else %}{{ page.url | remove_first:'/' }}{% endif %}">
```

Comment:
Let's improve og-tags code. Also `page.description` is not correct value for og:description in case of product page - should be product.description (NB! html must be removed).

```liquid
{%- if article -%}
  {%- assign og_obj = article -%}
{%- elsif product -%}
  {%- assign og_obj = product -%}
{%- else -%}
  {%- assign og_obj = page -%}
{%- endif -%}
...

<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first: "/" }}">

...

{%- if og_obj.image? -%}
  {%- assign og_image = page.image.for-width-1200 -%}
{%- endif -%}

{%- if og_obj.description != blank -%}
  <meta property="og:description" content="{{ og_obj.description | strip_html | escape_once }}">
  <meta name="description" content="{{ og_obj.description | strip_html | escape_once }}">
{%- endif -%}
```